addons/mail/static/src/models/thread/thread.js

```diff
@@ -1296,6 +1296,18 @@ function factory(dependencies) {
             this._currentPartnerLastNotifiedIsTyping = isTyping;
         }
 
+        /**
+         * Handles change of messages on the current thread. In particular,
+         * instructs currently displayed caches to refresh.
+         */
+        _onMessagesChanged() {
+            for (const threadView of this.threadViews) {
+                if (threadView.threadCache) {
+                    threadView.threadCache.update({ isCacheRefreshRequested: true });
+                }
+            }
+        }
+
```

Comment:
This might still not be the best idea. We'll see at least have one recursion of result of message fetch itself triggering this.

Probably this should be made flux specific, just like fold and pin. Flux would be reception of new message on the bus?

Reply:
What are the conditions/flux on which we want to force refresh the currently displayed filtered thread?

Reply:
I'll move this onchange to thread cache itself and only trigger it for the main cache, that should be better.